test/UnitTests/MSTest.Analyzers.UnitTests/DoNotUseShadowingAnalyzerTests.cs

```diff
@@ -0,0 +1,164 @@
+            {
+                public void Method(int x) { }
+            }
+
+            [TestClass]
+            public class DerivedClass : BaseClass
+            {
+                public void [|Method|](int y) { }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenTestClassHaveOverrideProperty_NoDiagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            public abstract class BaseClass
+            {
+                public abstract int Method { get; }
+            }
+
+            [TestClass]
+            public class DerivedClass : BaseClass
+            {
+                public override int Method { get; }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenTestClassOverrideMethodFromBaseClass_NoDiagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            public abstract class BaseClass
+            {
+                public abstract void Method();
+            }
+
+            [TestClass]
+            public class DerivedClass : BaseClass
+            {
+                public override void Method() { }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenTestClassHaveShadowingProperty_Diagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            public class BaseClass
+            {
+                public TestContext TestContext { get; } = null;
+            }
+
+            [TestClass]
+            public class DerivedClass : BaseClass
+            {
+                public new TestContext [|TestContext|] { [|get|]; } = null;
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenTestClassHaveShadowingMethod_WithoutNewModifier_Diagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            public class BaseClass
+            {
+                public void Method() { }
+            }
+
+            [TestClass]
+            public class DerivedClass : BaseClass
+            {
+                public void [|Method|]() { }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenTestClassHaveShadowingProperty_WithoutNewModifier_Diagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            public class BaseClass
+            {
+                public TestContext TestContext { get; set; } = null;
+            }
+
+            [TestClass]
+            public class DerivedClass : BaseClass
+            {
+                public TestContext [|TestContext|] { [|get|]; [|set|]; } = null;
```

Comment:
We probably want only one case reported here (you will need to filter out accessor methods).